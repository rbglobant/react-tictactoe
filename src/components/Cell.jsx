const Cell = ({ value, onClick }) => {
    return ( 
        <button className="game-container_board-cell" onClick={onClick}>
            {value}
        </button> );
}
 
export default Cell;