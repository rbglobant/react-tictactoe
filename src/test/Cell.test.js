import React from "react";
import Enzyme from "enzyme";
import Cell from "../components/Cell";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
Enzyme.configure({ adapter: new Adapter() });

describe("<Cell />", () => {

  test("Cell populates with the passed prop value", () => {
    const wrapper = Enzyme.shallow(<Cell value={1}/>);
    expect(wrapper.html()).toMatchSnapshot();
  });
  
  test("Cell callback function is called on click", () => {
    console.log = jest.fn();
    const wrapper = Enzyme.shallow(<Cell value={1} onClick={() => {console.log('click')}}/>);
    wrapper.find(".game-container_board-cell").simulate("click");
    expect(console.log).toHaveBeenCalledWith('click');
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
