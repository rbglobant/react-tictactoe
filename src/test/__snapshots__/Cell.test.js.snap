// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> Cell populates with the passed prop value 1`] = `"<button class=\\"game-container_board-cell\\">1</button>"`;
